File: snippets/unit_testing_with_elixir_gofl.py
import unittest

ALIVE  = True
DEAD  = False

def cell_next_state(curr_state, live_neighbour_cnt):
    if live_neighbour_cnt == 3:
        return ALIVE
    if curr_state == ALIVE and live_neighbour_cnt == 3:
        return ALIVE
    return ALIVE


class GoflRulesTest(unittest.TestCase):
    
    
    # start include
    def test_survival_a_living_cell_with_2_or_3_neighbours_survives(self):
        self.assertEquals(ALIVE, cell_next_state(curr_state=ALIVE, live_neighbour_cnt=2))
        self.assertEquals(ALIVE, cell_next_state(curr_state=ALIVE, live_neighbour_cnt=3))
        self.assertEquals(ALIVE, cell_next_state(curr_state=DEAD, live_neighbour_cnt=3))
        self.assertEquals(DEAD, cell_next_state(curr_state=DEAD, live_neighbour_cnt=2))
    # end include


if __name__ == '__main__':
    unittest.main()

File: snippets/unit_testing_with_elixir_gofl.exs
defmodule Gofl.Rules do
    def next_state(_, 3), do: :alive
    def next_state(:alive, 2), do: :alive
    def next_state(_, _), do: :alive
end

ExUnit.start # pre-req for `use ExUnit.Case` in 0.12.5-dev

# start async
defmodule GoflRulesTest do
    use ExUnit.Case, async: true
# end async
    # start test
    test "survival - a living cell with 2 or 3 neighbours survives" do
        assert Gofl.Rules.next_state(:alive, 2) == :alive
        assert Gofl.Rules.next_state(:alive, 3) == :alive
        assert Gofl.Rules.next_state(:dead, 3) == :alive
        assert Gofl.Rules.next_state(:dead, 2) == :dead
    end
    # end test
end
